File: ros_learning_pkg/ros_learning_pkg/smart_phone.py
#! /usr/bin python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class SmartphoneSubscriber(Node):
    def __init__(self):
        super().__init__("Smartphone")
        
        self.subscriber_ = self.create_subscription(String,
                                    "news_station_pub", self.callback_news_pub,10)
        self.get_logger().info("Smartphone node started...")
        
    def callback_news_pub(self,msg):
        self.get_logger().info(msg.data)
        
        
def main(args= None):
    
    rclpy.init(args= args)
    node = SmartphoneSubscriber()
    rclpy.spin(node)
    rclpy.shutdown()
    
if __name__ == "main":
    main()

File: ros_learning_cpp/src/smart_phone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class SmartphoneSubscriber:public rclcpp::Node{
    public:
    SmartphoneSubscriber():Node("Smartphone"){
        subscriber_ = this->create_subscription<example_interfaces::msg::String>("news_station_pub",10,
                        std::bind(&SmartphoneSubscriber::callback_msg_news,this,
                        std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(),"Smartphone node started...");

    }
    private:
    
    void callback_msg_news(const example_interfaces::msg::String::SharedPtr msg){
        RCLCPP_INFO(this->get_logger(),"%s",msg->data.c_str());
    }
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartphoneSubscriber>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}